tools/tuf/README.md

```diff
@@ -0,0 +1,292 @@
+    github("<div><img src='../../website/assets/images/github-mark-white-24x24@2x.png' /></div>Github Action\n(signing and notarization)");
+    
+    usb--(1) copy encrypted signing keys-->releaser;
+    1password--(2) get passphrases to decrypt encrypted signing keys-->releaser;
+    1password--(3) get Github API token-->releaser;
+    s3--(4) pull TUF repository-->releaser;
+    releaser--(5) build components (new updates)\n(osqueryd, orbit, Fleet Desktop)-->github;
+    github--(6) download built components-->releaser;
+    releaser--(7) push updates and signed metadata-->s3;
+```
+
+## Permissions and configuration
+
+Following is the checklist for all credentials and configuration needed to run the script.
+
+### Dependencies
+
+- `make`
+- `git`
+- 1Password 8 application.
+- Install and configure 1Password's `op` cli to connect to the application: https://developer.1password.com/docs/cli/get-started/
+- `aws` cli :`brew install awscli`.
+- `fleetctl`: Either built from source or installed by npm.
+- `tuf`: Download the release from https://github.com/theupdateframework/go-tuf/releases/download/v0.7.0/tuf_0.7.0_darwin_amd64.tar.gz and place the `tuf` executable in `/usr/local/bin/tuf`. You will need to make an exception in "Privacy & Security" because the executable is not signed.
+
+### 1Password
+
+You need to create three passphrases on your private 1Password vault for encrypting the signing keys (more on signing keys below).
+Create three private "passwords" with the following names: `TUF TARGETS`, `TUF SNAPSHOT` and `TUF TIMESTAMP`.
+The resulting credentials will have the following "path" within 1Password (these paths will be provided to the `releaser.sh` script)
+```sh
+Private/TUF TARGETS/password
+Private/TUF SNAPSHOT/password
+Private/TUF TIMESTAMP/password
+```
+
+### AWS
+
+The following is required to be able to run `aws` cli commands.
+
+1. You will need to request the infrastructure team to add the "TUFAdministrators" role to your Google account.
+2. Configure AWS SSO with the following steps: https://github.com/fleetdm/confidential/tree/main/infrastructure/sso#how-to-use-sso.
+Set the profile name as `tuf` (the profile name will be provided to the `releaser.sh` script).
+3. Test the access by running:
+```sh
+AWS_PROFILE=tuf aws sso login
+```
+
+### TUF signing keys
+
+> You can skip this step if you already have authorized keys to sign and publish updates.
+
+To release updates to our TUF repository you need the `root` role (ask in Slack who has such `root` role) to sign your signing keys.
+First, run the following script
+```sh
+AWS_PROFILE=tuf \
+ACTION=generate-signing-keys \
+TUF_DIRECTORY=/Users/luk/tuf3.fleetctl.com \
+TARGETS_PASSPHRASE_1PASSWORD_PATH="Private/TUF TARGETS/password" \
+SNAPSHOT_PASSPHRASE_1PASSWORD_PATH="Private/TUF SNAPSHOT/password" \
+TIMESTAMP_PASSPHRASE_1PASSWORD_PATH="Private/TUF TIMESTAMP/password" \
+./tools/tuf/releaser.sh
```

Comment:
> So, 2 people can run this flow -- someone else + root?

Correct, since these are newly authorized keys they need to be manually signed in an air gap environment booted from USB to an older, trusted version of Linux.

Reply:
Correct. It involves two humans.

- The human requesting their (targets, snapshot and timestamp) keys be signed runs this command.
- The human with the root role runs the below steps to sign the keys.

(This only needs to be performed once.)